lib/ProMotion/table/table.rb

```diff
@@ -30,13 +30,15 @@ def promotion_table_data
       @promotion_table_data ||= TableData.new(table_data, table_view)
     end
 
-    def set_up_header_view
-      if self.respond_to?(:table_header_view)
-        header_view = self.table_header_view
-        if header_view.is_a? UIView
-          self.tableView.tableHeaderView = header_view
-        else
-          PM.logger.warn "Table header views must be a UIView."
+    def set_up_header_footer_views
+      [:header, :footer].each do |hf_view|
+        if self.respond_to?("table_#{hf_view}_view".to_sym)
```

Comment:
Don't change this, but just fyi, you can also do:

```ruby
if self.respond_to?(:"table_#{hf_view}_view"))
```

In this case, a string works just as well as a symbol too.